deploy.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+quit() {
+    if [ $1 -eq 1 ]; then
+        echo "ERROR: "$2
+    else
+        echo $2
+    fi
+    exit $1
+}
+
+if [ ! -z $1 ]; then
+    UPDATE=$1
+else
+    quit 1 "Undefinied parameter. 1 for install, 0 to update."
+fi
+
+if [ ! -z $2 ]; then
+    DEST_DIR=$2
+else
+    quit 1 "Undefined DEST_DIR"
+fi
+
+FED_KEY=""
+MAIN_BLOCK_NUMBER=""
+SIDE_BLOCK_NUMBER=""
+ETH_HOST=""
+RSK_HOST=""
+
+cd $DEST_DIR/federator
+
+config() {
+    cp $DEST_DIR/federator/config/config.sample.js $DEST_DIR/federator/config/config.js &&
+        sed -i "s|rsktestnet-kovan|rskmainnet|g" $DEST_DIR/federator/config/config.js &&
+        sed -i "s|kovan|ethmainnet|g" $DEST_DIR/federator/config/config.js &&
+        return 0
+    quit 1 "There was a problem setting the configuration, please verify."
+}
+
+dependencies() {
+    echo "Installing required packages ..."
+    npm install &&
+        echo "Packages installed" &&
+        return 0
+    quit 1 "There was an error installing the needed packages, please check"
+}
+
+key() {
+    while [ ! -f $DEST_DIR/federator/config/federator.key ]; do
+        read -p "Please move your private key to $DEST_DIR/federator/config/federator.key, then press any key to continue " FED_KEY
+        continue
+    done
+    echo ""
+    echo "Private key placed on $DEST_DIR/federator/config/federator.key"
+    return 0
+}
+
+block_exists() {
+    grep -q '^[0-9]*$' "$DEST_DIR/federator/db/lastBlock.txt" 2>/dev/null &&
+        grep -q '^[0-9]*$' "$DEST_DIR/federator/db/side-fed/lastBlock.txt" 2>/dev/null &&
+        return 0
+    return 1
+}
+
+eth_host(){
```

Comment:
Do we want to make this non-interactive? Asking for a configuration value presents the same problem as we for the federator private key.

Reply:
In lieu of the rest of the code and its evolution, I'll consider the answer is "no". Will mark this as resolved.